### PR TITLE
feat: [CI-7836]: gitness integration - connectorRef as a non required field for …

### PR DESCRIPTION
**Pull request by shaurya-harness**
### Summary

gitness integration - connectorRef as a non required field for codebase
this PR takes care of : 
1. while creating/editing a pipeline connectorRef will be an optional field if clone codebase is enabled, at BE if repoName is provided but connectorRef is not then they will assume to fetch from gitness
2. in case codebase is set as runtime input then connectorRef will also be optional for triggers/input sets
3. the above points will also apply to templates with build stage
4. currently fetchDefaultBranchForRepo api is only called if we have a connectorRef in the yaml, this behaviour will change to fetch the default branch without connectorRef (if not provided)

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/70-pipeline/components/PipelineInputSetForm/CICodebaseInputSetForm.tsx [hunk 1]
```diff
@@ -839,52 +834,49 @@ function CICodebaseInputSetFormInternal({
...
                       )}
-                      {(isBuildRuntimeInput ||
-                        (isConnectorRuntimeInput && get(formik?.values, codeBaseInputFieldFormName.connectorRef))) &&
-                        !showBuildAsDisabledTextField && (
-                          <Layout.Horizontal
-                            flex={{ justifyContent: 'start' }}
-                            padding={{ top: 'small', left: 'xsmall', bottom: 'xsmall' }}
-                            margin={{ left: 'large' }}
-                          >
+                      {isBuildRuntimeInput && !showBuildAsDisabledTextField && (
```
Comment (d1wilko) [inline src/modules/70-pipeline/components/PipelineInputSetForm/CICodebaseInputSetForm.tsx:837]: I think we should still show these fields if the connector is a runtime input?

File: src/modules/70-pipeline/components/PipelineInputSetForm/CICodebaseInputSetForm.tsx [hunk 2]
```diff
@@ -839,52 +834,49 @@ function CICodebaseInputSetFormInternal({
...
-                        )}
+                          )}
+                        </Layout.Horizontal>
+                      )}
                       {codebaseTypeError && (formik.submitCount > 0 || viewTypeMetadata?.isTrigger) && (
                         <Text color={Color.RED_600}>{codebaseTypeError}</Text>
                       )}
-                      {(buildIncludesRuntimeInput ||
-                        (isConnectorRuntimeInput && get(formik?.values, codeBaseInputFieldFormName.connectorRef))) && (
+                      {buildIncludesRuntimeInput && (
```
Comment (d1wilko) [inline src/modules/70-pipeline/components/PipelineInputSetForm/CICodebaseInputSetForm.tsx:879]: I think these should also show if connector is runtime input?

**Review by d1wilko — CHANGES_REQUESTED** [review]
mostly looks good - but some questions

File: src/modules/70-pipeline/components/PipelineInputSetForm/CICodebaseInputSetForm.tsx (hunk 2 above)
Comment (shaurya-harness) [reply src/modules/70-pipeline/components/PipelineInputSetForm/CICodebaseInputSetForm.tsx:879]: yes adding it back

File: src/modules/70-pipeline/components/PipelineInputSetForm/CICodebaseInputSetForm.tsx (hunk 1 above)
Comment (shaurya-harness) [reply src/modules/70-pipeline/components/PipelineInputSetForm/CICodebaseInputSetForm.tsx:837]: yes adding it back

**Pull request closed without merging**